phys/module_diag_trad_fields.F

```diff
@@ -40,20 +42,27 @@ SUBROUTINE trad_fields ( u,v,w,t,qv,zp,zb,pp,pb,p,pw,            &
                                                                          f,e,sina,cosa,ht
       INTEGER, INTENT(IN   )                                          :: use_theta_m
       REAL   , INTENT(IN   ) , DIMENSION(ims:ime , kms:kme , jms:jme) :: u,v,w,t,qv,zp,zb,pp,pb,p,pw
+
+      REAL   , INTENT(IN   ) , DIMENSION(ims:ime , kms:kme , jms:jme) :: qc, rho, dz8w
+      REAL   , INTENT(IN   ) , DIMENSION(ims:ime , jms:jme)           :: psfc, rainc, rainnc, hailnc,graupelnc, snownc
    
       !  Output variables
    
-      REAL   , INTENT(  OUT) ,  DIMENSION(ims:ime , jms:jme)           :: sealevelp
       REAL   , INTENT(  OUT) ,  DIMENSION(ims:ime , kms:kme , jms:jme) :: temperature , &
                                                                           pressure    , &
                                                                           geoheight   , &
                                                                           umet        , &
                                                                           vmet        , &
                                                                           speed       , &
+                                                                          potential_t , &
+                                                                          rh          , &
                                                                           dir
+      REAL   , INTENT(  OUT) ,  DIMENSION(ims:ime , jms:jme)           :: sealevelp, rain, liqrain,tpw
+      REAL   , INTENT(  OUT) ,  DIMENSION(ims:ime , jms:jme)           :: psfc2d
    
       !  Local variables
    
+      REAL :: ptot
       REAL,  DIMENSION(its:ite , jts:jte) :: pscf
```

Comment:
@smileMchen 
Ming,
The field is named pscf, not psfc. The psfc field cannot be used in a computation.

Reply:
@Dave,
pscf is an variable in the original code and it is never used. I think we should delete this variable. 
psfc is passed into the subroutine "trad_fields" and its value is given to psfc2d. We use the variable psfc2d to represent psfc in the output file for traditional diagnostic variables.